backend/gisst-cli/src/main.rs

```
@@ -200,6 +200,7 @@ async fn add_patched_instance(
             .await?;
         }
     }
+    tx.commit().await;
```

Nice; I think it would be good to either throw a ? on the end (and add an error variant for converting from a transaction failure if needed) or even unwrap it, since we don't want silently ignored errors.

---

Done! I converted the error to a GISSTCliError::Sql since I wasn't sure whether to create a new error type for the transactions or not.

---

Good choice!

---

I like the idea here but as you see it does not compile.  You probably want `tx.commit().await.map_err(GISSTCliError::Sql)`.  If you put it in a closure it means you need to use `(|e| GISSTCliError::Sql(e))`.  The error you’re getting says “you have a function from a SQL error to a GISSTCliError, but I can’t convert that to a GISSTCliError”.

---

Ah I see, so the conversion from GISSTCLIError needs to be defined already. Do you know what type "e" was before it tried to convert into GISSTCliError::Sql?

---

e was a SqlxError, so that conversion is fine; you could have written GISSTCliError:SQL(e) inside the closure, but since the closure took one argument and passed it to the variant constructor it’s cleaner to just use the variant constructor GISSTCliError::SQL directly.

In Haskell or Coq it’s like writing:
`map (\x -> double x) l` vs `map double l`.  In your earlier version you had the equivalent of `map (\x -> double) l`, which doesn’t apply the argument and doesn’t have the type you wanted.